### PR TITLE
Adds the `array_prepend(value, array)` scalar and concat operator

### PR DESCRIPTION
**Pull request by seut**
The ``value || array`` concat operator is an alias to the ``array_prepend`` scalar.

### CODE REVIEW

File: server/src/main/java/io/crate/expression/scalar/ArrayPrependFunction.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    public final List<Object> evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input[] args) {
+        ArrayList<Object> resultList = new ArrayList<>();
+        Object valueToAdd = args[0].value();
+        @SuppressWarnings("unchecked")
+        List<Object> values = (List<Object>) args[1].value();
+
+        if (valueToAdd != null || calledByOperator == false) {
+            // null || array -> array (null is ignored)
+            resultList.add(innerType.sanitizeValue(valueToAdd));
+        }
```
Comment (seut) [inline server/src/main/java/io/crate/expression/scalar/ArrayPrependFunction.java:83]: The opposite (not going into the if branch) is already tested by the existing null literal concat operator test at https://github.com/crate/crate/blob/master/server/src/test/java/io/crate/expression/scalar/ConcatFunctionTest.java#L129. This matches now the new array_prepend` scalar.

File: server/src/main/java/io/crate/expression/scalar/ArrayPrependFunction.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+    @Override
+    public final List<Object> evaluate(TransactionContext txnCtx, NodeContext nodeCtx, Input[] args) {
+        ArrayList<Object> resultList = new ArrayList<>();
+        Object valueToAdd = args[0].value();
+        @SuppressWarnings("unchecked")
+        List<Object> values = (List<Object>) args[1].value();
+
+        if (valueToAdd != null || calledByOperator == false) {
+            // null || array -> array (null is ignored)
```
Comment (matriv) [inline server/src/main/java/io/crate/expression/scalar/ArrayPrependFunction.java:81]: I think, I find it more clear if the comment is moved above the if.

File: docs/general/builtins/scalar-functions.rst [hunk 3]
```diff
@@ -3239,6 +3239,40 @@ Begin the search from given position (optional).
...
+    cr> select
+    ...    1 || [2,3,4] AS array_prepend;
+    +---------------+
+    | array_prepend |
+    +---------------+
+    | [1, 2, 3, 4]  |
+    +---------------+
+    SELECT 1 row in set (... sec)
+
+
```
Comment (matriv) [inline docs/general/builtins/scalar-functions.rst:3275]: could we add a NOTE about the difference in null handling between function call and operator?

**Review by matriv — COMMENTED** [review]
Thank you! left 2 minor comments.

File: docs/general/builtins/scalar-functions.rst (hunk 3 above)
Comment (seut) [reply docs/general/builtins/scalar-functions.rst:3287]: done, see https://github.com/crate/crate/pull/16333/commits/aaff072332ac8698cba9ac5b4e264b175315f367

**Pull request merged**